syncd/syncd.cpp

```diff
@@ -2782,9 +2782,11 @@ void processFlexCounterGroupEvent(
 {
     SWSS_LOG_ENTER();
 
-    std::lock_guard<std::mutex> lock(g_mutex);
     swss::KeyOpFieldsValuesTuple kco;
-    consumer.pop(kco);
+    {
```

Comment:
why you moved this here? please revert this change, since g_mutex is not protecting only consumer.pop but other variables too!!